### PR TITLE
Migrate Gauge component from Plotly to PatternFly 4

### PR DESCRIPTION
**Pull request by TheRealJon**
Replace plotly.js gauge implementation with PF4 ChartDonutThreshold and ChartDonutUtilization. 

**Cluster Overview**
image

**Resource Quota Details**
image

### CODE REVIEW

File: frontend/public/components/utils/units.js [hunk 1]
```diff
@@ -101,19 +101,12 @@ const convertValueWithUnitsToBaseValue = (value, unitArray, divisor) => {
   return { value, unit };
 };
 
-const round = units.round = (value) => {
+const round = units.round = (value, decimalPlace) => {
```
Comment (spadgett) [inline frontend/public/components/utils/units.js]: If we're reworking this, we should consider `Intl.NumberFormat` to format correctly for the user's locale:

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/NumberFormat

We use it for the overview:

https://github.com/spadgett/console/blob/a69288f379eb9817b2fc798c977ad28d17a6a3a5/frontend/public/components/overview/project-overview.tsx#L27

It has handling for formatting as percent as well, which I see you added below.

File: frontend/public/components/graphs/prometheus-graph.tsx [hunk 2]
```diff
@@ -18,19 +18,19 @@ export const getPrometheusExpressionBrowserURL = (urls, queries): string => {
...
   }
 );
 
-export const PrometheusGraph: React.FC<PrometheusGraphProps> = React.forwardRef(({children, className, query, title}, ref: React.RefObject<HTMLDivElement>) => {
+export const PrometheusGraph: React.FC<PrometheusGraphProps> = React.forwardRef(({children, className, title}, ref: React.RefObject<HTMLDivElement>) => {
   return <div ref={ref} className={classNames('graph-wrapper', className)}>
-    { title && <h5 className="graph-title graph-title--left">{title}</h5> }
-    <PrometheusGraphLink query={query}>{children}</PrometheusGraphLink>
+    { title && <h5 className="graph-title">{title}</h5> }
+    {children}
```
Comment (spadgett) [inline frontend/public/components/graphs/prometheus-graph.tsx]: style nit: inconsistent spacing inside braces on these lines (we need an eslint rule)

**Comment by TheRealJon** [comment]
@spadgett I've addressed your feedback

File: frontend/public/components/graphs/gauge.tsx [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+  invert = false,
+  label = '',
+  namespace,
+  percent = 0,
+  remainderLabel = 'available',
+  query = '',
+  thresholds = DEFAULT_THRESHOLDS,
+  title,
+  usedLabel = 'used',
+}: GaugeProps) => {
```
Comment (kyoto) [inline frontend/public/components/graphs/gauge.tsx]: Shouldn't need to specify `GaugeProps` twice

File: frontend/public/components/graphs/gauge.tsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+  },
+  {
+    value: 92,
+    color: ThresholdColor.ERROR,
+  },
+];
+
+const handleResponse = (response: PrometheusResponse): number => {
+  const v = _.get(response, 'data.result[0].value', [0,0]);
+  return parseFloat(v[1]);
```
Comment (kyoto) [inline frontend/public/components/graphs/gauge.tsx]: Could this be `parseFloat(_.get(response, 'data.result[0].value[1]', 0))`?

**Comment by kyoto** [comment]
Thanks @TheRealJon, looks great!

I added two very minor comments, but LGTM.

File: frontend/public/components/utils/units.js [hunk 5]
```diff
@@ -101,20 +96,27 @@ const convertValueWithUnitsToBaseValue = (value, unitArray, divisor) => {
...
   }
-  const multiplier = Math.pow(10, digits);
-  return Math.round(value * multiplier) / multiplier;
+  return 1;
+};
+
+const round = units.round = (value, options) => {
+  const fractionDigits = getDefaultFractionDigits(value);
+  const {locales, ...rest} = _.defaults(options, {
+    locales: 'en-US',
```
Comment (spadgett) [inline frontend/public/components/utils/units.js]: I don't think we want to default locale. If we leave it out, the system locale is used.

> The locales argument must be either a string holding a BCP 47 language tag, or an array of such language tags. If the locales argument is not provided or is undefined, the runtime's default locale is used.

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl#Locale_identification_and_negotiation

Comment (TheRealJon) [reply frontend/public/components/utils/units.js]: Fixed

**Pull request merged**